support/cas-server-support-syncope-authentication/src/main/java/org/apereo/cas/syncope/pm/SyncopePasswordManagementService.java

```diff
@@ -0,0 +1,135 @@
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apereo.cas.authentication.Credential;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.pm.PasswordChangeRequest;
+import org.apereo.cas.pm.PasswordHistoryService;
+import org.apereo.cas.pm.PasswordManagementQuery;
+import org.apereo.cas.pm.impl.BasePasswordManagementService;
+import org.apereo.cas.syncope.SyncopeUtils;
+import org.apereo.cas.util.crypto.CipherExecutor;
+import org.apereo.cas.util.function.FunctionUtils;
+import org.apereo.cas.util.http.HttpExecutionRequest;
+import org.apereo.cas.util.http.HttpUtils;
+import org.apereo.cas.util.serialization.JacksonObjectMapperFactory;
+import org.apereo.cas.util.spring.SpringExpressionLanguageValueResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+
+public class SyncopePasswordManagementService extends BasePasswordManagementService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SyncopePasswordManagementService.class);
+
+    private static final ObjectMapper MAPPER = JacksonObjectMapperFactory.builder()
+            .defaultTypingEnabled(false).build().toObjectMapper();
+
+    public SyncopePasswordManagementService(final CipherExecutor<Serializable, String> cipherExecutor,
+                                            final CasConfigurationProperties casProperties,
+                                            final PasswordHistoryService passwordHistoryService) {
+        super(casProperties, cipherExecutor, passwordHistoryService);
+    }
+
+    @Override
+    public boolean changeInternal(final PasswordChangeRequest bean) {
+        return FunctionUtils.doAndHandle(() -> {
+            String syncopeRestPasswordResetUrl = StringUtils.appendIfMissing(
+                    SpringExpressionLanguageValueResolver.getInstance().resolve(
+                            casProperties.getAuthn().getPm().getSyncope().getUrl()),
+                    "/rest/users/self/mustChangePassword");
+            LOGGER.debug("Updating account password on syncope for user [{}]", bean.getUsername());
+            HttpExecutionRequest exec = HttpExecutionRequest.builder()
+                    .method(HttpMethod.POST)
+                    .url(syncopeRestPasswordResetUrl)
+                    .basicAuthUsername(bean.getUsername())
+                    .basicAuthPassword(bean.toCurrentPassword())
+                    .headers(Map.of("X-Syncope-Domain", casProperties.getAuthn().getSyncope().getDomain(),
+                            HttpHeaders.ACCEPT, "application/json",
+                            HttpHeaders.CONTENT_TYPE, "application/json"))
+                    .entity(MAPPER.writeValueAsString(getPasswordPatch(bean)))
+                    .maximumRetryAttempts(1)
```

Comment:
Retry attempts should be configurable; I think there is a setting for this already that was recently added, right?